tests/python/pants_test/engine/test_fs.py

```diff
@@ -359,6 +360,44 @@ def test_merge_directories(self):
 
       self.assertEqual(both_snapshot.directory_digest, both_merged)
 
+  def test_asynchronously_merge_directories(self):
+    with temporary_dir() as temp_dir:
+      with open(os.path.join(temp_dir, "roland"), "w") as f:
+        f.write("European Burmese")
+      with open(os.path.join(temp_dir, "susannah"), "w") as f:
+        f.write("Not sure actually")
+      scheduler = self.mk_scheduler(rules=create_fs_rules())
```

Comment:
Since this extends `TestBase`, I think that you should be able to use `self.scheduler`? Fine either way though.

Reply:
Done